### DISCUSSION TITLE
Misorientation and KAM

### DISCUSSION THREAD

**Discussion by NBKIZ** [mtex-toolbox/mtex#1996] [Ask Anything]
Hello everyone,
  I am trying to overlay KAM plot with the misorienation profile/plot. Is there any way I could do it? I tried different ways but there seems to be some issue with the color bar and the plot doesnt come up. In brief, I want a KAM plot with the misorienation grain boundaries indicated.

Thanks in advance

**Comment by kilir** [comment]
Hi,
I'm not entirely sure what you mean by overlaying a map plot and a profile plot, but simply grain boundary misorientation angle on top of kam would be:
```
mtexdata titanium
plot(ebsd,ebsd.KAM/degree)
hold on
grains = ebsd.calcGrains
plot(grains.boundary,grains.boundary.misorientation.angle/degree)
hold off
mtexColorbar
```

**Reply by NBKIZ** [reply]
Hello,
  Thanks for the answer. I was actually trying to get KAM in form of color bar and mirorientation of grain boundaries in line colour and thats where I was facing an error.
% Plot KAM data
figure;
h_kam = plot(ebsd, ebsd.KAM('threshold', 5*degree) ./ degree);
colorbar; % Add a colorbar for KAM data
hold on;
% Select grain boundaries for a specific phase (e.g., 'Ti-Hex' to 'Ti-Hex')
gB = grains.boundary;
gB_MgMg = gB('Ti-Hex', 'Ti-Hex');
% Plot grain boundaries with color based on misorientation angles
angle_thresholds = [5, 15, 50, 70];
colors = {'r', 'g', 'b', 'm'};
% Use antipodal option for correct orientation
plot(gB_MgMg, gB_MgMg.misorientation.angle ./ degree, 'linewidth', 2, 'antipodal', 'DisplayName', 'Grain Boundaries');

% Apply color to grain boundaries based on misorientation angles
for i = 1:length(angle_thresholds)-1
    indices = gB_MgMg.misorientation.angle >= angle_thresholds(i) & gB_MgMg.misorientation.angle < angle_thresholds(i+1);
    gB_MgMg(indices).set('color', colors{i});
end

% Display colorbar for misorientation-based coloring
mtexColorbar;

hold off;

But still I am not able to correctly align the legends and get the graph
